### PR TITLE
[RFC] Zero padding can be removed using torch-scatter

### PR DESCRIPTION
**Pull request by CompRhys**
I have updated as little as possible of the code in order to remove zero-padding from the model. This means that the code can now consider more than a fixed number of neighbours without introducing new 'zero padding' atoms. 

Such 'zero padding' atoms are potentially a problem as the use of affine transformations means that the 'zero padding' atoms affect the final result as the bias vectors will introduce no zero features vectors for these atoms.

The _torch_scatter_ library has been used to perform the necessary scatter operations.

The pre-trained models have not been re-trained.

feel free to make changes to this PR.

### CODE REVIEW

**Comment by CompRhys** [comment]
Closed in favour of #18.
